ipa-core/src/protocol/hybrid/oprf.rs

```diff
@@ -122,9 +131,31 @@ where
     .try_collect::<Vec<_>>()
     .await?;
 
-    let prf_key = gen_prf_key(&ctx.narrow(&IpaPrfStep::PrfKeyGen));
+    // In order to compute the same PRF value for the same match key
+    // all shards must have the same prf_key. This has the leader
+    // generate a key, and give it to all the followers.
+    let prf_key_ctx = ctx.set_total_records(TotalRecords::ONE);
+    let prf_key = if ctx.shard_id() == ShardIndex::FIRST {
+        let leader_prf_key = gen_prf_key(&ctx.narrow(&HybridStep::PrfKeyGen));
+
+        for shard in ctx.shard_count().iter().skip(1) {
+            prf_key_ctx
+                .clone()
+                .shard_send_channel::<Replicated<Fp25519>>(shard)
+                .send(RecordId::FIRST, leader_prf_key.clone())
+                .await?;
+        }
+        Ok(leader_prf_key)
+    } else {
+        let mut receiver = prf_key_ctx
+            .shard_recv_channel::<Replicated<Fp25519>>(ShardIndex::FIRST)
+            .take(1);
+        receiver.next().await.unwrap()
+    }
+    .unwrap();
```

Comment:
I couldn't seem to get this to work with ? instead of the unwrap()s here, which is not great. any suggestions on this would be helpful.

that said, this will get replaced with #1410, so maybe work either just waiting on that, or accepting this until it can be replaced.

Reply:
#1410 is merged so we can replace this I think